src/components/PopoverMenu.tsx

```diff
@@ -122,8 +120,8 @@ function PopoverMenu({
             const selectedSubMenuItemIndex = selectedItem?.subMenuItems.findIndex((option) => option.isSelected);
             setFocusedIndex(selectedSubMenuItemIndex);
         } else {
-            selectedItemIndex.current = index;
             onItemSelected(selectedItem, index);
+            selectedItem.onSelected?.();
```

Comment:
I'm trying to understand this component, why do we have two callbacks when we click a menu item? `onItemSelected` / `selectedItem.onSelected`?

Reply:
You can ignore that question, I see that they are just two "types" of callback, and in the broken flow `onItemSelected(selectedItem, index)` takes care of closing the context menu (menu opened with the 3 dots), meanwhile `selectedItem.onSelected?.();` is supposed to open the next confirmation modal.

I think the difference in the implementation is that now `selectedItem.onSelected?.();` gets called earlier. Before it was called in the `onModalHide` callback.